src/type.js

```diff
@@ -324,6 +326,11 @@ function typeCharacter(
         newEntry = textToBeTyped
       }
 
+      const timeNewEntry = buildTimeValue(textToBeTyped)
+      if (isValidInputTimevalue(currentElement(), timeNewEntry)) {
```

Comment:
```suggestion
      if (isValidInputTimeValue(currentElement(), timeNewEntry)) {
```

Reply:
Applied here https://github.com/testing-library/user-event/pull/502/commits/32b5dd248184898055a1f5ee2ec5ace134add290 (same for the other comments)